Sources/HAP/Endpoints/characteristics().swift

```diff
@@ -65,11 +113,18 @@ func characteristics(device: Device) -> Application {
                     .flatMap({$0.characteristics.filter({$0.iid == iid})})
                     .first else
                 {
-                    return .notFound
+                    return .invalidParameters
                 }
 
                 // set new value
                 if let value = item["value"] {
+                    guard characteristic.permissions.contains(.write) else {
+                        logger.info("\(characteristic) has no write permission")
+                        serialized.append(["aid": aid,"iid": iid,"status": HAPStatusCodes.readOnly.rawValue])
```

Comment:
Please put a space after the comma for readability.

Reply:
Fixed